modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -136,8 +148,17 @@ public Settings getAdditionalIndexSettings(
      * Alternatively this method can instead parse mappings into map of maps and merge that and
      * iterate over all values to find the field that can serve as routing value. But this requires
      * mapping specific logic to exist here.
+     *
+     * In addition to static mappings, dynamic templates can also contain fields annotated as
+     * time_series_dimension. These fields are identified and added to a separate list, if one is provided.
      */
-    private List<String> findRoutingPaths(String indexName, Settings allSettings, List<CompressedXContent> combinedTemplateMappings) {
+    private void findRoutingPathsAndDimensions(
```

Comment:
Maybe keep the existing `findRoutingPaths` method unchanged and create a `findDimensions(...)` method?
This keeps the two modes (index routing path & dynamic dimensions) separate.

Reply:
I've splitted this in three functions to reuse the MappingService, ptal.